metadata-ingestion/src/datahub/ingestion/source/iceberg/__init__.py

```diff
@@ -0,0 +1,341 @@
+                    LOGGER.exception(f"Exception while processing table {self.config.filesystem_url}/{tablePath.name}, skipping it.", e)
+
+    def _createIcebergWorkunit(self, datasetName: str, table: Table) -> Iterable[MetadataWorkUnit]:
+        self.report.report_table_scanned(datasetName)
+        datasetUrn = make_dataset_urn(self.platform, datasetName, self.config.env)
+        dataset_snapshot = DatasetSnapshot(
+            urn=datasetUrn,
+            aspects=[],
+        )
+
+        customProperties = dict(table.properties())
+        customProperties["location"] = table.location()
+        customProperties["snapshot-id"] = str(table.current_snapshot().snapshot_id)
+        customProperties["manifest-list"] = table.current_snapshot().manifest_location
+        dataset_properties = DatasetPropertiesClass(
+            tags=[],
+            customProperties=customProperties,
+        )
+        dataset_snapshot.aspects.append(dataset_properties)
+
+        # TODO: The following seems to go in OperationClass.  It is usually part of the "usage" source.  Should I create a new python module?
+        last_updated_time_millis=table.current_snapshot().timestamp_millis # Should we use last-updated-ms instead?  YES
+        operation_aspect = OperationClass(
+            timestampMillis=get_sys_time(),
+            lastUpdatedTimestamp=last_updated_time_millis,
+            operationType=OperationTypeClass.UNKNOWN,
+        )
+        # TODO: How do I add an operation_aspect?
+        
+        if "owner" in table.properties():
+            ownerEmail = table.properties()["owner"]
+            owners = [OwnerClass(
+                owner=f"urn:li:corpuser:{ownerEmail}",
+                type=OwnershipTypeClass.PRODUCER,
+                source=None,
+            )]
+            dataset_ownership = OwnershipClass(
+                owners=owners
+            )
+            dataset_snapshot.aspects.append(dataset_ownership)
+
+        schemaMetadata = self._createSchemaMetadata(datasetName, table)
+        dataset_snapshot.aspects.append(schemaMetadata)
+
+        mce = MetadataChangeEvent(proposedSnapshot=dataset_snapshot)
+        wu = MetadataWorkUnit(id=datasetName, mce=mce)
+        self.report.report_workunit(wu)
+        yield wu
+
+        if self.config.profiling.enabled:
+            profiler = IcebergProfiler(self.report, self.config.profiling)
+            yield from profiler.profileTable(self.config.env, datasetName, table, schemaMetadata)
+
+    def _createSchemaMetadata(self, dataset_name: str, table: Table) -> SchemaMetadata:
+        schema_fields = self.get_schema_fields(dataset_name, table.schema().columns())
+
+        schema_metadata = SchemaMetadata(
+            schemaName=dataset_name,
+            platform=make_data_platform_urn(self.platform),
+            version=0,
+            hash="",
+            platformSchema=SchemalessClass(), # TODO: Not sure what to use here...
```

Comment:
If the schema is undefined, we've used an `OtherSchemaClass` before – you can find it in schema_classes.py

Reply:
What's the role of a `platformSchema`?  I have no clue...